### DISCUSSION TITLE
Create3 control through the web

### DISCUSSION THREAD

**Discussion by jdbrubaker** [iRobotEducation/create3_docs#272] [Questions]
Hi, I would like to undock, dock, and control the robot through a web interface.  I have tried calling a PHP file on my web server -shell_exec('undock.sh).  My undock.sh script calls the ros2 undock action.  This works fine when undock.sh is run fro the terminal, but does nothing through php.  Is there any solution?  thanks ---john.

**Comment by shamlian** [comment]
This doesn't seem to be a problem with the robot, but I have some debugging suggestions which may or may not help. From your question, it seems like the web server can run ROS 2 and send the proper commands to execute actions, Can you provide the source of `undock.sh` ? Perhaps something is not getting set properly for ROS 2 to start. Also, what RMW are you using, and what version of ROS 2 are you running on the server and on the robot?
